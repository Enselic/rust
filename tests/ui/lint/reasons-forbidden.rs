#![feature(lint_reasons)]

#![forbid(
    unsafe_code,
    //~^ NOTE `forbid` level set here
    //~| NOTE the lint level is defined here
    reason = "our errors & omissions insurance policy doesn't cover unsafe Rust"
)]

use std::ptr;

fn main() {
    let a_billion_dollar_mistake = ptr::null();

    #[allow(unsafe_code)]
    //~^ ERROR allow(unsafe_code) incompatible with previous forbid
    //~| NOTE our errors & omissions insurance policy doesn't cover unsafe Rust
    //~| NOTE overruled by previous forbid
    unsafe {
        //~^ ERROR usage of an `unsafe` block
        *a_billion_dollar_mistake
    }
}
