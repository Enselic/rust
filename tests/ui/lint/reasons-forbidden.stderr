error[E0453]: allow(unsafe_code) incompatible with previous forbid
  --> $DIR/reasons-forbidden.rs:15:13
   |
LL |     unsafe_code,
   |     ----------- `forbid` level set here
...
LL |     #[allow(unsafe_code)]
   |             ^^^^^^^^^^^ overruled by previous forbid
   |
   = note: our errors & omissions insurance policy doesn't cover unsafe Rust

error[E0453]: allow(unsafe_code) incompatible with previous forbid
  --> $DIR/reasons-forbidden.rs:15:13
   |
LL |     unsafe_code,
   |     ----------- `forbid` level set here
...
LL |     #[allow(unsafe_code)]
   |             ^^^^^^^^^^^ overruled by previous forbid
   |
   = note: our errors & omissions insurance policy doesn't cover unsafe Rust
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: usage of an `unsafe` block
  --> $DIR/reasons-forbidden.rs:19:5
   |
LL | /     unsafe {
LL | |
LL | |         *a_billion_dollar_mistake
LL | |     }
   | |_____^
   |
   = note: our errors & omissions insurance policy doesn't cover unsafe Rust
note: the lint level is defined here
  --> $DIR/reasons-forbidden.rs:4:5
   |
LL |     unsafe_code,
   |     ^^^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0453`.
